bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/bus/KNXBinding.java

```diff
@@ -108,7 +108,7 @@ public void removeKNXTypeMapper(KNXTypeMapper typeMapper) {
     @Override
     protected void internalReceiveCommand(String itemName, Command command) {
         logger.trace("Received command (item='{}', command='{}')", itemName, command.toString());
-        if (!isEcho(itemName, command)) {
+        if (!KNXConnection.getCompatibilityOH2() && !isEcho(itemName, command)) {
```

Comment:
Your second question:
https://github.com/openhab/openhab/blob/1.7/bundles/binding/org.openhab.binding.knx/OSGI-INF/knxgenericbindingprovider.xml
https://github.com/openhab/openhab/blob/1.8/bundles/binding/org.openhab.binding.knx/OSGI-INF/knxgenericbindingprovider.xml

No, it seams AutoUpdateBindingProvider has been never been active! 
I have not realized this over the years, what I would have spared. :-(